lib/src/domain/location/get_location_use_case.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:allthenews/src/domain/location/location.dart';
+import 'package:allthenews/src/domain/location/location_exception.dart';
+import 'package:allthenews/src/domain/location/location_resolver.dart';
+
+class GetLocationUseCase {
+  final LocationResolver _locationResolver;
+
+  GetLocationUseCase(this._locationResolver);
+
+  Future<Location> call() async {
+    final isServiceEnabled = await _locationResolver.isLocationServiceEnabled();
+    if (!isServiceEnabled) {
+      return Future.error(LocationServiceDisabledException());
+    }
+
+    if (await _locationResolver.isPermissionDeniedForever()) {
+      return Future.error(PermissionDeniedForeverException());
```

Comment:
obecnie taki jest rezultat przy wejściu do mapy, mam włączoną lokalizację na urządzeniu. Imo powinno pytać usera o permission w tym miejscu i pokazywać error tylko jeżeli odrzuci prośbę
![Uploading location.gif…]()

Reply:
możliwe, że to błąd związany z tym, że isPermissionDenied i isPermissionDeniedForever sprawdzały to samo, a nie miały, teraz chyba będzie ok, przynajmniej 'u mnie działa' :P